File: GanttChartLightLibraryDemos/Demos/Samples/WPF-CSharp/GanttChartDataGrid/StatusColumns/StatusGanttChartItem.cs
using DlhSoft.Windows.Controls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Media;

namespace Demos.WPF.CSharp.GanttChartDataGrid.StatusColumns
{
    class StatusGanttChartItem : GanttChartItem
    {
        public string Status
        {
            get
            {
                if (HasChildren || IsMilestone)
                    return string.Empty;
                if (CompletedFinish >= Finish)
                    return "Completed";
                var now = DateTime.Now;
                if (CompletedFinish < now)
                    return "Behind schedule";
                if (CompletedFinish > Start)
                    return "In progress";
                return "To Do";
            }
        }

        public SolidColorBrush StatusColor
        {
            get
            {
                switch (Status)
                {
                    case "Completed":
                        return Brushes.Green;
                    case "To Do":
                        return Brushes.Gray;
                    case "Behind schedule":
                        return Brushes.Red;
                    case "In progress":
                        return Brushes.Orange;
                    default:
                        return Brushes.Transparent;
                }
            }
        }

        protected override void OnPropertyChanged(string propertyName)
        {
            base.OnPropertyChanged(propertyName);

            var statusAffectingPropertyNames = new[] { nameof(Start), nameof(CompletedFinish), nameof(Finish), nameof(IsMilestone), nameof(HasChildren) };
            if (statusAffectingPropertyNames.Contains(propertyName))
            {
                OnPropertyChanged(nameof(Status));
                OnPropertyChanged(nameof(StatusColor));
            }
        }
    }
}

File: GanttChartLightLibraryDemos/Demos/Samples.Resources/WPF-VisualBasic/GanttChartDataGrid/StatusColumns/StatusGanttChartItem.vb
Imports DlhSoft.Windows.Controls
Imports System.ComponentModel
Imports System.Text

Friend Class StatusGanttChartItem
    Inherits GanttChartItem

    Public ReadOnly Property Status() As String
        Get
            If HasChildren OrElse IsMilestone Then
                Return String.Empty
            End If
            If CompletedFinish >= Finish Then
                Return "Completed"
            End If
            Dim now = Date.Now
            If CompletedFinish < now Then
                Return "Behind schedule"
            End If
            If CompletedFinish > Start Then
                Return "In progress"
            End If
            Return "To Do"
        End Get
    End Property

    Public ReadOnly Property StatusColor() As SolidColorBrush
        Get
            Select Case Status
                Case "Completed"
                    Return Brushes.Green
                Case "To Do"
                    Return Brushes.Gray
                Case "Behind schedule"
                    Return Brushes.Red
                Case "In progress"
                    Return Brushes.Orange
                Case Else
                    Return Brushes.Transparent
            End Select
        End Get
    End Property

    Protected Overrides Sub OnPropertyChanged(propertyName As String)
        MyBase.OnPropertyChanged(propertyName)

        Dim statusAffectingPropertyNames = {NameOf(Start), NameOf(CompletedFinish), NameOf(Finish), NameOf(IsMilestone), NameOf(HasChildren)}
        If statusAffectingPropertyNames.Contains(propertyName) Then
            OnPropertyChanged(NameOf(Status))
            OnPropertyChanged(NameOf(StatusColor))
        End If
    End Sub
End Class